src/main/data/TwoBit.js

```diff
@@ -124,6 +136,32 @@ function markUnknownDNA(basePairs: Array<string>, dnaStartIndex: number, sequenc
   return basePairs;
 }
 
+/**
+ * Try getting a bigger chunk of the remote file
+ * until the Range Error is resolved. This is useful when we need to
+ * parse the header, but when we don't know the size of the header up front.
+ * If the intial request returns an incomplete header and hence the
+ * parsing fails, we next try doubling the requested size.
+ * The size of the request is capped with `untilSize` so that
+ * we don't start asking for MBs of data for no use.
+ * Instead we we throw an error if we reach the cap,
+ * potentially meaning a corrupted 2bit file.
+*/
+function retryRemoteGet(remoteFile: RemoteFile, start: number, size: number, untilSize: number, promisedFunc: Function) {
+  return remoteFile.getBytes(start, size).then(promisedFunc).catch(error => {
+    if(error.constructor == RangeError) {
+      // Do not attempt to download more than `untilSize`
+      if(size > untilSize) {
+        throw `Couldn't parse the header ` +
+              `from the first ${size} bytes of the file. ` +
+              `Corrupted 2bit file?`;
+      }
+      return retryRemoteGet(remoteFile, start, size*2, untilSize, promisedFunc);
```

Comment:
That is definitely possible, but that solution would require passing around partial buffers and concatenating them, which is not that hard to implement; but, I thought the way it is right now is  easier to read/interpret and cleaner. Normally, I don't expect this logic to be called that often, which also made me think that wasting tens or hundreds of KBs for rare edge cases shouldn't be that big of a problem.

So yeah — I chose readability over performance, but if you think that might eventually bite us in a way I don't foresee, I am happy to optimize those network requests.